pkg/sparrow/run.go

```diff
@@ -137,118 +141,93 @@ func (s *Sparrow) Run(ctx context.Context) error {
 
 // ReconcileChecks registers new Checks, unregisters removed Checks,
 // resets the Configs of Checks and starts running the checks
-func (s *Sparrow) ReconcileChecks(ctx context.Context) {
-	for name, checkCfg := range s.cfg.Checks {
-		name := name
-		log := logger.FromContext(ctx).With("name", name)
-
-		c := s.updateCheckTargets(checkCfg)
-		if existingCheck, ok := s.checks[name]; ok {
-			// Check already registered, reset config
-			err := existingCheck.SetConfig(ctx, c)
+func (s *Sparrow) ReconcileChecks(ctx context.Context, cfg runtime.Config) {
+	// generate checks from configuration
+	s.enrichTargets(cfg)
+	fChecks, err := factory.NewChecksFromConfig(cfg)
```

Comment:
The check's targets are never enriched since you've forgot use the return value of `enrichTargets(cfg)`

Reply:
Additionally write a unit test for this in the reconcile testing

Reply:
Great catch, I've added the a mock gitlab target manager to the run_test + a DNS test